models/exposure.py

```diff
@@ -250,21 +250,42 @@ def format(self, value):
 
     exp_time = sa.Column(sa.REAL, nullable=False, index=True, doc="Exposure time in seconds. ")
 
-    filter = sa.Column(sa.Text, nullable=True, index=True, doc="Name of the filter used to make this exposure. ")
+    _filter = sa.Column(sa.Text,
+                        nullable=True, #nullable because a constraint checks _filter or filter_array
+                        index=True,
+                        doc=("Name of the filter used to make this exposure. "
+                             "This is generally the short filter name - conversion "
+                             "to the proper name for each instrument happens in its code."))
 
     airmass = sa.Column(sa.REAL, nullable=True, index=True, doc="Airmass taken from the header of the exposure. ")
 
     @property
-    def filter_short(self):
-        if self.filter is None:
+    def filter( self ):
+        if self._filter is None:
             return None
-        return self.instrument_object.get_short_filter_name(self.filter)
+        else:
+            return self._filter
+
+    @filter.setter
+    def filter( self, val ):
+        # Once instrument can only be a valid instrument, should add
+        # a check here which says "IF there is an instrument, only
+        # allow a valid filter for that instrument". Leaving the
+        # ability to set any filter at the moment for tests.
```

Comment:
Make an issue to go along with this comment.

Reply:
#391